clang/lib/DPCT/SaveNewFiles.cpp

```diff
@@ -156,16 +156,26 @@ void rewriteFileName(clang::tooling::UnifiedPath &FileName,
   // SPT_CudaSource, SPT_CppSource and SPT_CudaHeader files.
   if (DpctGlobalInfo::getChangeExtensions().empty() ||
       DpctGlobalInfo::getChangeExtensions().count(Extension.str())) {
-    if (FileType & SPT_CudaSource)
+    if (FileType & SPT_CudaSource) {
       path::replace_extension(CanonicalPathStr,
                               DpctGlobalInfo::getSYCLSourceExtension());
-    else if (FileType & SPT_CppSource)
-      path::replace_extension(CanonicalPathStr,
-                              Extension +
-                                  DpctGlobalInfo::getSYCLSourceExtension());
-    else if (FileType & SPT_CudaHeader)
+    } else if (FileType & SPT_CppSource) {
+      if (Extension == ".c") {
```

Comment:
Please also consider  ".C" or a function:  is_c_code()

Reply:
The .C(capital letter) usually means C++ source file extension.
https://gcc.gnu.org/onlinedocs/gcc/Invoking-G_002b_002b.html